libs/langchain/langchain/chains/combine_documents/reduce.py

```diff
@@ -2,13 +2,166 @@
 
 from __future__ import annotations
 
-from typing import Any, Callable, List, Optional, Protocol, Tuple
+from functools import partial
+from typing import Any, Callable, Dict, List, Optional, Protocol, Sequence, Tuple, Union
 
 from langchain_core.documents import Document
+from langchain_core.language_models import LanguageModelLike
+from langchain_core.output_parsers import StrOutputParser
+from langchain_core.prompts import BasePromptTemplate, PromptTemplate
 from langchain_core.pydantic_v1 import Extra
+from langchain_core.runnables import (
+    Runnable,
+    RunnableLambda,
+    RunnableParallel,
+    RunnablePassthrough,
+)
 
 from langchain.callbacks.manager import Callbacks
-from langchain.chains.combine_documents.base import BaseCombineDocumentsChain
+from langchain.chains.combine_documents.base import (
+    DEFAULT_DOCUMENT_SEPARATOR,
+    DOCUMENTS_KEY,
+    BaseCombineDocumentsChain,
+    format_document_inputs,
+    format_documents,
+    validate_prompt,
+)
+
+""" --- LCEL Runnable chains --- """
+
+
+def create_collapse_documents_chain(
+    llm: LanguageModelLike,
+    prompt: BasePromptTemplate,
+    *,
+    token_max: int,
+    document_prompt: Optional[BasePromptTemplate] = None,
+    document_separator: str = DEFAULT_DOCUMENT_SEPARATOR,
+    token_len_func: Optional[Callable[[str], int]] = None,
+) -> Runnable[Dict[str, Any], List[Document]]:
+    """Create
+
+    Args:
+        llm: Language model to use for collapsing document contents.
+        prompt: Prompt to use for collapsing document contents. Must accept "context" as
+            one of the input variables. The formatted documents will be passed in as
+            "context".
+        token_max: The maximum cumulative token length that the list of Documents can
```

Comment:
no default?

Reply:
seems like the kind of thing you'd almost always want to set yourself (since it depends on model), so default would obfuscate more than anything?